homeassistant/components/homekit/type_fans.py

```diff
@@ -0,0 +1,159 @@
+        self._flag[CHAR_ACTIVE] = True
+
+        service = SERVICE_TURN_ON if value == 1 else SERVICE_TURN_OFF
+        self.hass.services.call(self._domain, service,
+                                {ATTR_ENTITY_ID: self.entity_id})
+
+    @debounce
+    def set_speed(self, value):
+        """Set speed if call came from HomeKit."""
+        _LOGGER.debug('%s: Set speed to %d', self.entity_id, value)
+        self._flag[CHAR_ROTATION_SPEED] = True
+        if value != 0:
+            speed_list = self.hass.states.get(self.entity_id) \
+                .attributes.get(ATTR_SPEED_LIST)
+            speed = speed_list[int(round(value / self.min_step)) - 1]
+            self.hass.components.fan.turn_on(
+                self.entity_id, speed=speed)
+        else:
+            self.hass.components.fan.turn_off(self.entity_id)
+
+    def set_direction(self, value):
+        """Set state if call came from HomeKit."""
+        _LOGGER.debug('%s: Set direction to %d', self.entity_id, value)
+        self._flag[CHAR_ROTATION_DIRECTION] = True
+        direction = HOMEKIT_TO_HASS[value]
+        self.hass.components.fan.set_direction(self.entity_id,
+                                               direction=direction)
+
+    def set_oscillating(self, value):
+        """Set state if call came from HomeKit."""
+        _LOGGER.debug('%s: Set oscillating to %d', self.entity_id, value)
+        self._flag[CHAR_SWING_MODE] = True
+        oscillating = True if value == 1 else False
+        self.hass.components.fan.oscillate(self.entity_id,
+                                           should_oscillate=oscillating)
+
+    def update_state(self, new_state):
+        """Update fan after state change."""
+        # Handle State
+        state = new_state.state
+        if state in (STATE_ON, STATE_OFF):
+            self._state = 1 if state == STATE_ON else 0
+            if not self._flag[CHAR_ACTIVE] and \
+                    self.char_active.value != self._state:
+                self.char_active.set_value(self._state)
+            self._flag[CHAR_ACTIVE] = False
+
+        # Handle Speed
+        if CHAR_ROTATION_SPEED in self.chars:
+            speed_list = new_state.attributes.get(ATTR_SPEED_LIST)
+            speed = new_state.attributes.get(ATTR_SPEED)
+            if not self._flag[CHAR_ROTATION_SPEED] and \
+                    len(speed_list) > 1 and isinstance(speed, str):
+                self._speed = (speed_list.index(speed) + 1) * self.min_step
+                if self.char_speed.value != self._speed:
+                    self.char_speed.set_value(self._speed)
+            self._flag[CHAR_ROTATION_SPEED] = False
+
+        # Handle Direction
+        if CHAR_ROTATION_DIRECTION in self.chars:
+            direction = new_state.attributes.get(ATTR_DIRECTION)
+            if not self._flag[CHAR_ROTATION_DIRECTION] and \
+                    direction in HASS_TO_HOMEKIT:
+                self._direction = HASS_TO_HOMEKIT[direction]
```

Comment:
Same as before. I think `if` is more than enough here.

Reply:
No problem, updated.